src/_pytest/monkeypatch.py

```diff
@@ -290,12 +290,17 @@ def delattr(
             self._setattr.append((target, name, oldval))
             delattr(target, name)
 
-    def setitem(self, dic: MutableMapping[K, V], name: K, value: V) -> None:
+    def setitem(self, dic: Mapping[K, V], name: K, value: V) -> None:
         """Set dictionary entry ``name`` to value."""
         self._setitem.append((dic, name, dic.get(name, notset)))
         dic[name] = value
 
-    def delitem(self, dic: MutableMapping[K, V], name: K, raising: bool = True) -> None:
+    def delitem(
+        self,
+        dic: Mapping[K, V],
+        name: K,
+        raising: bool = True,
+    ) -> None:
```

Comment:
Let's keep this unchanged:

```suggestion
    def delitem(self, dic: Mapping[K, V], name: K, raising: bool = True) -> None:
```